### PR TITLE
Support templates in directories several levels deep

### PR DESCRIPTION
**Pull request by mrship**
I have a view template structure in my Rails app that goes several directories deep. 

To support that with cache_digests, this patch alters the `#directory` method in `CacheDigests::TemplateDigestor`to pop the template name off the end of the path and return the full view directory structure instead of taking just the first directory.

I'm sure I won't be the only one with views nested more than one directory deep, so can this be considered please. Thanks.

### CODE REVIEW

File: lib/cache_digests/template_digestor.rb [hunk 1]
```diff
@@ -68,7 +68,9 @@ def logical_name
       end
       
       def directory
-        name.split("/").first
+        split = name.split("/")
+        split.pop
+        split.join("/")
```
Comment (dhh) [inline lib/cache_digests/template_digestor.rb]: Inline this as `name.split("/")[0..-2].join("/")`.

**Comment by dhh** [comment]
Just one comment on the implementation. Otherwise looks good. Please prepare a PR for rails/master as well. Thanks for finding and fixing this.

**Comment by mrship** [comment]
OK, I've inlined the directory splitting as suggested. I'll do the rails PR shortly.

**Comment by mrship** [comment]
I'm having issues with getting Rails to install from source locally. It's hanging on the `bundle install` when communicating with Rubygems. However, I've patched [my fork](https://github.com/mrship/rails) of Rails with the changes needed. 

Perhaps you can review the [commit](https://github.com/mrship/rails/commit/2938ef7a65be5527905174b348b58c98a06d20e4) and see if it's enough for you to commit to rails/master? I appreciate it isn't ideal, but I'd be unsure about creating a PR for rails/master that I couldn't actually test myself.

Thoughts?

**Comment by mrship** [comment]
OK, I got it working (eventually!). [PR](https://github.com/rails/rails/pull/7912) for Rails submitted.

**Comment by dhh** [comment]
Perfect, thanks again.

**Pull request merged**